src/auto_note.cpp

```diff
@@ -0,0 +1,454 @@
+                }
+            } else if( name == "discovered" ) {
+                jin.start_array();
+                while( !jin.end_array() ) {
+                    const std::string entry = jin.get_string();
+                    this->discovered.insert( entry );
+                }
+            } else {
+                jin.skip_value();
+            }
+
+        }
+    };
+
+    // Actually read save file
+    if( !read_from_file_optional_json( this->build_save_path(), parseJson ) ) {
+        // If loading from JSON failed, fall back to default initialization
+        this->default_initialize();
+        this->save();
+    }
+}
+
+void auto_note_settings::default_initialize()
+{
+    // Perform clear for good measure
+    this->clear();
+
+    // Fetch all know map extra types and iterate over them
+    for( auto extra : MapExtras::mapExtraFactory().get_all() ) {
+        if( extra.autonote ) {
+            this->autoNoteEnabled.insert( extra.id.str() );
+        }
+    }
+}
+
+void auto_note_settings::set_discovered( const string_id<map_extra> &mapExtId )
+{
+    this->discovered.insert( mapExtId.str() );
+}
+
+bool auto_note_settings::was_discovered( const string_id<map_extra> &mapExtId ) const
+{
+    return this->discovered.count( mapExtId.str() ) != 0;
+}
+
+void auto_note_settings::show_gui()
+{
+    // Create local GUI instance and show
+    auto_note_manager_gui gui{ };
+    gui.show();
+
+    // If settings were changed, we need to do a save
+    if( gui.was_changed() ) {
+        this->save();
+    }
+}
+
+bool auto_note_settings::has_auto_note_enabled( const std::string &mapExtId ) const
+{
+    return this->autoNoteEnabled.count( mapExtId ) != 0;
+}
+
+void auto_note_settings::set_auto_note_status( const std::string &mapExtId, const bool enabled )
+{
+    if( enabled ) {
+        this->autoNoteEnabled.insert( mapExtId );
+    } else if( this->has_auto_note_enabled( mapExtId ) ) {
+        this->autoNoteEnabled.erase( mapExtId );
+    }
+}
+
+bool auto_note_settings::has_auto_note_enabled( const string_id<map_extra> &mapExtId ) const
+{
+    return this->has_auto_note_enabled( mapExtId.str() );
+}
+
+void auto_note_settings::set_auto_note_status( const string_id<map_extra> &mapExtId,
+        const bool enabled )
+{
+    this->set_auto_note_status( mapExtId.str(), enabled );
+}
+
+void auto_note_manager_gui::initialize()
+{
+    // Clear all data
+    this->mapExtraCache.clear();
+
+    // Fetch reference to current auto note settings
+    const auto_note_settings &settings = get_auto_notes_settings();
+
+    // Fetch all know map extra types and iterate over them
+    for( auto extra : MapExtras::mapExtraFactory().get_all() ) {
+        // Ignore all extras that have autonote disabled in the JSON.
+        // This filters out lots of extras users shouldnt see (like "normal")
+        if( !extra.autonote ) {
+            continue;
+        }
+
+        // Check if auto note is enabled for this map extra type
```

Comment:
Many of your comments are just repeating the code itself - why? `has_auto_note_enabled` describes what it does quite well. Even more obvious is the comment above `mapExtraCache.clear();` - yes, `clear` clears the data, I had assumed that even without the comment.

Comments explaining **why** something happens (like for the `if( !extra.autonote )`) or how something works are fine.